### ISSUE TITLE
[Autowiring] Controller/Actions are not autowired if called through twig render*( 

### ISSUE THREAD

**Issue by Plopix** [symfony/symfony#35909]
**Symfony version(s) affected**: 5.0
(it works on 4.4)

**Description**  
Autowiring is correctly set up and it is working as expected but not when using `render` functions.
https://github.com/Plopix/symfony-autowiring-render-twig-bug/blob/master/sf5/src/Controller/DefaultController.php

OK in SF 4.4, not on SF 5.0

**How to reproduce**  
Here is a reproducer.
https://github.com/Plopix/symfony-autowiring-render-twig-bug

just  checkout
```
cd sf4
composer install
symfony server:start
```

```
cd sf5
composer install
symfony server:start
```

> but you'd know that ;)

**Comment by noemi-salaun** [comment]
You issue come from your controller syntax `App\Controller\DefaultController:embed`. It should be `App\Controller\DefaultController::embed`. Note the double colon.

Since SF 4.1 the first notation is deprecated.
See https://github.com/symfony/symfony/blob/master/UPGRADE-4.1.md#frameworkbundle

The change takes place in `\Symfony\Component\HttpKernel\DependencyInjection\RemoveEmptyControllerArgumentLocatorsPass`.

### SF 4.4
```php
                if (!$reason) {
                    // Deprecated since Symfony 4.1. See Symfony\Component\HttpKernel\Controller\ContainerControllerResolver
                    $controllers[$id.':'.$action] = $argumentRef;

                    if ('__invoke' === $action) {
                        $controllers[$id] = $argumentRef;
                    }
                    continue;
                }
```

### SF 5.0
```php
                if (!$reason) {
                    if ('__invoke' === $action) {
                        $controllers[$id] = $argumentRef;
                    }
                    continue;
                }
```

And easier reproductible case was just defining a route in yaml

```yaml
index:
    path: /
    controller: App\Controller\DefaultController:index    # <--- with 1 colon
```

I take a look at the deprecation logs, but using the wrong syntax in 4.4 does not trigger any deprecation :/

**Comment by Plopix** [comment]
hello @noemi-salaun,
Thanks for the investigation!

But, where did you see `::` in my code?

```twig
 {{ render(controller('App\\Controller\\DefaultController:embed', {params: {id: 13, something: 'plop'}})) }}
```

It is only one `:` in my reproducer.

Did I miss something?

**Comment by Plopix** [comment]
Nice @noemi-salaun! Indeed! it is not a bug then...That is interesting the syntax with `:` is still working!

**Comment by nicolas-grekas** [comment]
Shall we close then?

**Comment by Plopix** [comment]
I guess so. I mean 4.1 is a long time ago. If it's happening to someone they may find this closed issue for help.

**Comment by nicolas-grekas** [comment]
Fixed in #36017 actually.

**Comment by noemi-salaun** [comment]
@nicolas-grekas you can close my PR https://github.com/symfony/symfony/pull/36010 then. I didn't know if it was a bug in v5.0 or a missing deprecation in v4.1+. But apparently it was a bug ;)
